sway/tree/view.c

```diff
@@ -244,25 +261,32 @@ void view_autoconfigure(struct sway_view *view) {
 	double y_offset = 0;
 
 	if (!container_is_floating(con) && ws) {
-
 		bool smart = config->hide_edge_borders_smart == ESMART_ON ||
 			(config->hide_edge_borders_smart == ESMART_NO_GAPS &&
 			!gaps_to_edge(view));
-		bool hide_smart = smart && view_ancestor_is_only_visible(view);
 
 		if (config->hide_edge_borders == E_BOTH
-				|| config->hide_edge_borders == E_VERTICAL || hide_smart) {
+				|| config->hide_edge_borders == E_VERTICAL) {
 			con->border_left = con->x != ws->x;
 			int right_x = con->x + con->width;
 			con->border_right = right_x != ws->x + ws->width;
 		}
+
 		if (config->hide_edge_borders == E_BOTH
-				|| config->hide_edge_borders == E_HORIZONTAL || hide_smart) {
+				|| config->hide_edge_borders == E_HORIZONTAL) {
 			con->border_top = con->y != ws->y;
 			int bottom_y = con->y + con->height;
 			con->border_bottom = bottom_y != ws->y + ws->height;
 		}
 
+		if (smart && view_ancestor_is_only_visible(view)) {
+			bool show_border = !view_is_only_visible_in_tab_or_stack(view);
+			con->border_left = show_border;
```

Comment:
I believe these should be `con->border_left &= show_border`. See #4698 for more information.

Reply:
Thanks, confirmed that the current code is misbehaving re: #4698 and that `&`-ing fixes it.